client/src/main/java/io/pravega/client/BatchClientFactory.java

```diff
@@ -43,10 +42,10 @@
      *
      * @param scope The scope of the stream.
      * @param config Configuration for the client.
+     * @param connectionFactory Connection for the client.
      * @return Instance of BatchClientFactory implementation.
      */
-    static BatchClientFactory withScope(String scope, ClientConfig config) {
-        val connectionFactory = new ConnectionFactoryImpl(config);
+    static BatchClientFactory withScope(String scope, ClientConfig config, ConnectionFactory connectionFactory) {
```

Comment:
This is a public interface and this change is backwards incompatible. We don't want users to have to add an additional step to get their client running.